src/tests/test_utils/common_test_utils/tests/utils_tests.cpp

```diff
@@ -101,4 +102,28 @@ TEST(UtilsTests, split_by_delimiter_empty_and_comma) {
     const std::unordered_set<std::string> expected_set = {"", ""};
 
     ASSERT_EQ(split_set, expected_set);
+}
+
+TEST(UtilsTests, device_monitor) {
+    std::string cpu_device_id = "";
+    std::map<std::string, float> utilization;
+    ASSERT_NO_THROW(utilization = get_device_utilization(cpu_device_id));
+#ifdef _WIN32
+    ASSERT_FALSE(utilization.empty() && utilization.count("Total") && utilization.at("Total") >= 0.0f)
+        << "Expected non-empty utilization map for CPU device";
+#else
+    bool ret = utilization == std::map<std::string, float>{{"Total", -1.0f}};
+    ASSERT_TRUE(ret) << "Expected utilization map with 'Total' key only for CPU device";
+#endif
+
+    std::string invalid_device_id = "INVALID_DEVICE_ID";
+#ifdef _WIN32
+    ASSERT_THROW(utilization = get_device_utilization(invalid_device_id), std::runtime_error)
```

Comment:
Why have different behaviour on different OS. 
It should be better to have unified and on windows catch exception handle internally and also not throw